File: Algorithms(easy)/121_Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        mi = prices[0]
        for i in prices[1:]:
            res = max(res, i - mi)
            mi = min(mi, i)
        return res
'''
mi用来找队列最小数，res找最大差，因为是顺序查找，不会存在反着减的问题
'''

File: Algorithms_C++/121_Best_Time_to_Buy_and_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        int mi = 1e9;
        for(int v : prices){
            ans = max(ans, v - mi);
            mi = min(mi, v);
        }
        return ans;
    }
};